[Task] Move sitemaps and redirects to SEO Bundle (#14348)
pimcore/pimcore: doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md

- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticateHttpBasic()` has been removed.
- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticatePlaintext()` has been removed.
- [DataObjects][CSV Export] Changed encoding of table data-types to `json_encode` from `base64_encoded`.

> - [Documents] The Message-Box whether a redirect should be created when a document is moved in the tree has been removed. This can now be set with the following configuration. This also counts for URLSlugs and pretty URLs

Reviewer: ```suggestion
- [Documents] Moving a document in the tree no longer opens the redirect prompt asking to create redirects. Creating a redirect is now configurable with pimcore:redirects:create_redirects. This config includes URLSlugs and Pretty URLs.
```

- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticateHttpBasic()` has been removed.
- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticatePlaintext()` has been removed.
- [DataObjects][CSV Export] Changed encoding of table data-types to `json_encode` from `base64_encoded`.

> - [Documents] Moving a document in the tree no longer opens the redirect prompt asking to create redirects. Creating a redirect is now configurable with `pimcore:redirects:create_redirects`. This config includes URLSlugs and Pretty URLs.

Reviewer 2: Also adapt SEO Bundle upgrade notes that Redirects & Sitemap moved into the bundle

Author: Done.

- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticateHttpBasic()` has been removed.
- [Authentication] Deprecated method `Pimcore\Tool\Authentication::authenticatePlaintext()` has been removed.
- [DataObjects][CSV Export] Changed encoding of table data-types to `json_encode` from `base64_encoded`.
- [Documents] Moving a document in the tree no longer opens the redirect prompt asking to create redirects. Creating a redirect is now configurable with `pimcore:redirects:auto_create_redirects`. This config includes URLSlugs and Pretty URLs.
```yaml
pimcore_seo:
    redirects:

>         create_redirect: true

Reviewer 2: ```suggestion
        auto_create_redirect: true
```

>         auto_create_redirect: true

Reviewer 2: ```suggestion
        auto_create_redirects: true
```
